### PR TITLE
Squash m-simonelli/file_transer onto blueprint/v3-2021-beta

### PR DESCRIPTION
**Pull request by pospeselr**
Squashes m-simonelli's file transfer commits ( https://github.com/m-simonelli/ricochet-refresh/tree/file_transfer ) and merges into one commit.

### CODE REVIEW

File: src/libtego/source/core/ConversationModel.cpp [hunk 1]
```diff
@@ -143,24 +184,30 @@ void ConversationModel::sendQueuedMessages()
...
-            return;
-        }
-    }
+    auto chat_channel = findOrCreateChannelForContact<Protocol::ChatChannel>(m_contact, Protocol::Channel::Outbound);
+    auto file_channel = findOrCreateChannelForContact<Protocol::FileChannel>(m_contact, Protocol::Channel::Outbound);
 
     // sendQueuedMessages is called at channelOpened
-    if (!channel->isOpened())
+    if (!chat_channel->isOpened())
         return;
```
Comment (m-simonelli) [inline src/libtego/source/core/ConversationModel.cpp:193]:
```
if (!file_channel->isOpened())
    return;
```

File: src/libtego/source/protocol/Connection.cpp [hunk 2]
```diff
@@ -180,6 +180,12 @@ void ConnectionPrivate::setSocket(QTcpSocket *s, Connection::Direction d)
         q->grantAuthentication(Connection::HiddenServiceAuth, serverName);
 
         // Send the introduction version handshake message
+        
+        /* XXX, since the 1.0.x legacy handling was dropped, introductions 
+         * should no longer say they support old versions */
+        /* Fix?:
+         * char intro[] = { 0x49, 0x4D, 0x01, ProtocolVersion }; */
+
```
Comment (m-simonelli) [inline src/libtego/source/protocol/Connection.cpp:188]: Solves #36

**Comment by m-simonelli** [comment]
Can you remove libtego/include/protocol/* please?

**Comment by pospeselr** [comment]
Ok, squashed over the latest with some minor changes (dropping whitespace changes, etc)

**Pull request merged**